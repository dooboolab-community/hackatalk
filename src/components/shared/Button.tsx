import {
  ActivityIndicator,
  ImageSourcePropType,
  TouchableOpacity,
  ViewStyle,
} from 'react-native';

import React from 'react';
import styled from 'styled-components/native';

interface StyledElement {
  white?: boolean;
}

interface ButtonContainer {
  width: number;
  height: number;
}

interface Props {
  testID: string;
  containerStyle?: ViewStyle;
  isWhite: boolean;
  isLoading: boolean;
  isDisabled: boolean;
  onPress: () => void;
  imgLeftSrc?: ImageSourcePropType;
  indicatorColor: string;
  activeOpacity: number;
  children: string;
  width: number;
  height: number;
}

const StyledButtonContainer = styled.View<ButtonContainer>`
  width: ${({ width }): string => `${width}px`};
  height: ${({ height }): string => `${height}px`};
  justify-content: center;
  align-items: center;
`;

const StyledButton = styled.View<StyledElement>`
  background-color: ${({ white, theme }): string =>
    white ? theme.btnPrimaryLight : theme.btnPrimary};
  border-color: ${({ white, theme }): string =>
    white ? theme.btnPrimary : theme.btnPrimaryLight};
  border-radius: 4px;
  border-width: 1px;
  align-items: center;
  justify-content: center;
`;

const StyledButtonDisabled = styled.View`
  background-color: rgb(243, 243, 243);
  align-self: center;
  border-radius: 4px;
  border-width: 2px;
  border-color: #333;
  align-items: center;
  justify-content: center;
`;

const StyledText = styled.Text<StyledElement>`
  font-size: 14px;
  font-weight: bold;
  color: ${({ white, theme }): string =>
    white ? theme.btnPrimaryLightFont : theme.btnPrimaryFont};
`;

const StyledImageLeft = styled.Image`
  width: 24px;
  height: 24px;
  position: absolute;
  left: 16px;
`;

function Button(props: Props): React.ReactElement {
  const {
    testID,
    isDisabled,
    isWhite,
    activeOpacity,
    onPress,
    isLoading,
    width,
    height,
    indicatorColor,
    imgLeftSrc,
    children,
  } = props;

  return (
    <>
      {isDisabled ? (
        <StyledButtonDisabled testID={testID}>
          <StyledText>{children}</StyledText>
        </StyledButtonDisabled>
      ) : (
        <TouchableOpacity
          testID={testID}
          activeOpacity={activeOpacity}
          onPress={onPress}
        >
          <StyledButton white={isWhite}>
            <StyledButtonContainer width={width} height={height}>
              {isLoading ? (
                <ActivityIndicator size="small" color={indicatorColor} />
              ) : imgLeftSrc ? (
                <StyledImageLeft source={imgLeftSrc} />
              ) : (
                undefined
              )}
              {!isLoading ? (
                <StyledText white={isWhite}>{children}</StyledText>
              ) : (
                undefined
              )}
            </StyledButtonContainer>
          </StyledButton>
        </TouchableOpacity>
      )}
    </>
  );
}

Button.defaultProps = {
  isWhite: false,
  isLoading: false,
  isDisabled: false,
  indicatorColor: 'white',
  activeOpacity: 0.5,
  width: 136,
  height: 60,
};

export default Button;
