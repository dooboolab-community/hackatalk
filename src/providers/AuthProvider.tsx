import React, { useReducer } from 'react';

import Relay from '../relay';
import { Environment as RelayEnvironment } from 'react-relay';
import { User } from '../types/graphql';
import createCtx from '../utils/createCtx';

interface Context {
  state: State;
  setUser(user: User | undefined): void;
}
const [useCtx, Provider] = createCtx<Context>();

export enum ActionType {
  SetUser = 'set-user',
}

export interface State {
  user: User | undefined | null;
  relay: RelayEnvironment;
}
type SetUserAction = {
  type: ActionType.SetUser;
  payload: { user: User; relay: RelayEnvironment };
};
type Action = SetUserAction;

interface Props {
  children?: React.ReactElement;
  initialAuthUser?: User;
}

type Reducer = (state: State, action: Action) => State;

const setUser = (dispatch: React.Dispatch<SetUserAction>) => (
  authUser: User,
): void => {
  if (!authUser) Relay.init();
  dispatch({
    type: ActionType.SetUser,
    payload: { user: authUser, relay: Relay.environment },
  });
};

const initialState: State = {
  user: undefined,
  relay: Relay.environment,
};

const reducer: Reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.SetUser:
      return { ...action.payload };
    default:
      return state;
  }
};

function AuthProvider(props: Props): React.ReactElement {
  const initialAuthState = {
    user: props.initialAuthUser,
    relay: Relay.environment,
  };
  const [state, dispatch] = useReducer<Reducer>(reducer, initialAuthState);

  const actions = {
    setUser: setUser(dispatch),
  };

  return <Provider value={{ state, ...actions }}>{props.children}</Provider>;
}

const AuthContext = {
  useAuthContext: useCtx,
  AuthProvider,
};

export { useCtx as useAuthContext, AuthProvider };
export default AuthContext;
