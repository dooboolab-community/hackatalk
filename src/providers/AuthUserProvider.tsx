import React, { useReducer } from 'react';

import { AuthUser } from '../types';
import createCtx from '../utils/createCtx';

interface Context {
  state: State;
  setAuthUser(authUser: AuthUser): void;
}
const [useCtx, Provider] = createCtx<Context>();

export enum ActionType {
  SetAuthUser = 'set-auth-user',
}

export interface State {
  user?: AuthUser;
}

type Action =
  | { type: ActionType.SetAuthUser; payload: State };

interface Props {
  children?: React.ReactElement;
  initialAuthUser?: AuthUser;
}

type Reducer = (state: State, action: Action) => State;

const setAuthUser = (dispatch: React.Dispatch<Action>) => (authUser: AuthUser): void => {
  dispatch({
    type: ActionType.SetAuthUser,
    payload: { user: authUser },
  });
};

const initialState: State = {};

const reducer: Reducer = (state = initialState, action) => {
  const { type, payload } = action;
  switch (type) {
    case ActionType.SetAuthUser:
      return {
        user: payload.user,
      };
    default:
      return state;
  }
};

function AuthUserProvider(props: Props): React.ReactElement {
  const { children, initialAuthUser } = props;
  const [state, dispatch] = useReducer<Reducer>(reducer, { user: initialAuthUser });

  const actions = {
    setAuthUser: setAuthUser(dispatch),
  };

  return <Provider value={{ state, ...actions }}>{children}</Provider>;
}

export { useCtx as useAuthUserContext, AuthUserProvider };
