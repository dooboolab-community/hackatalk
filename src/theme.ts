/* eslint-disable */
export const colors = {
  light: '#ffffff',
  dark: '#151A25',
  backgroundDark: '#151E22',
  deepDark: '#000000',
  blue_80: '#2F38C7',
  blue_70: '#1E6EFA',
  blue_60: '#217DFE',
  paleBlue_10: '#8bc9ff',
  paleBlue_05: '#e2f2ff',
  green_30: '#00D4AB',
  green_10: '#A1EEDB',
  gray_80: '#676c7a',
  gray_70: '#545966',
  gray_60: '#676c7a',
  gray_50: '#8f94a3',
  gray_40: '#AFB4C3',
  gray_30: '#d3d8e8',
  gray_05: '#f3f9ff',
  apple: '#151E22',
  google: '#E04238',
  facebook: '#345997',
};
/* eslint-enable */

export const light = {
  background: colors.light,
  backgroundDark: colors.paleBlue_05,
  itemBackground: colors.light,
  btnPrimary: colors.blue_70,
  btnPrimaryFont: colors.light,
  btnPrimaryLight: colors.light,
  btnPrimaryLightFont: colors.blue_70,
  textDisabled: colors.gray_50,
  btnDisabled: colors.gray_30,
  fontColor: colors.dark,
  fontSubColor: colors.blue_70,
  labelColor: colors.dark,
  tintColor: colors.green_30,
  lineColor: colors.gray_30,
  indicatorColor: colors.blue_70,
  inactiveColor: colors.gray_60,
  primary: colors.blue_70,
  primaryLight: colors.paleBlue_10,
  headerFont: colors.light,
  searchBackground: colors.gray_05,
  messageBackground: colors.light,
  status: colors.green_10,
  placeholder: colors.gray_60,
  focused: colors.blue_70,
  placeholderFocused: colors.blue_70,
  socialBtn: colors.paleBlue_05,
  link: colors.blue_70,
  myMessageBackground: colors.paleBlue_10,
  myMessageText: colors.dark,
  peerMessageBackground: colors.paleBlue_05,
  peerMessageText: colors.dark,
  modalBtnBackground: colors.light,
  modalBtnPrimaryFont: colors.blue_80,
  modalBtnFont: colors.gray_60,
  appleIcon: colors.apple,
  appleText: colors.apple,
  appleBackground: colors.light,
  facebookIcon: colors.light,
  facebookText: colors.light,
  facebookBackground: colors.facebook,
  googleIcon: colors.light,
  googleText: colors.light,
  googleBackground: colors.google,
  errorTitle: colors.gray_60,
  errorBody: colors.gray_50,
  fab: colors.blue_70,
};
export type Theme = typeof light;
export const dark = {
  background: colors.deepDark,
  backgroundDark: colors.deepDark,
  itemBackground: colors.dark,
  btnPrimary: colors.paleBlue_10,
  btnPrimaryFont: colors.deepDark,
  btnPrimaryLight: colors.dark,
  btnPrimaryLightFont: colors.paleBlue_10,
  textDisabled: colors.gray_80,
  btnDisabled: colors.gray_40,
  fontColor: colors.light,
  fontSubColor: colors.paleBlue_10,
  labelColor: colors.light,
  tintColor: colors.green_10,
  lineColor: colors.light,
  indicatorColor: colors.paleBlue_10,
  inactiveColor: colors.gray_50,
  primary: colors.deepDark,
  primaryLight: colors.dark,
  headerFont: colors.light,
  searchBackground: colors.gray_80,
  messageBackground: colors.dark,
  status: colors.green_30,
  placeholder: colors.gray_50,
  focused: '#ffffff',
  placeholderFocused: colors.paleBlue_10,
  socialBtn: colors.gray_05,
  link: colors.paleBlue_05,
  myMessageBackground: colors.deepDark,
  myMessageText: colors.light,
  peerMessageBackground: colors.paleBlue_05,
  peerMessageText: colors.dark,
  modalBtnBackground: colors.gray_80,
  modalBtnPrimaryFont: colors.paleBlue_10,
  modalBtnFont: colors.gray_40,
  appleIcon: colors.apple,
  appleText: colors.apple,
  appleBackground: colors.light,
  facebookIcon: colors.facebook,
  facebookText: colors.dark,
  facebookBackground: colors.light,
  googleIcon: colors.google,
  googleText: colors.dark,
  googleBackground: colors.light,
  errorTitle: colors.paleBlue_10,
  errorBody: colors.gray_40,
  fab: colors.paleBlue_10,
};
