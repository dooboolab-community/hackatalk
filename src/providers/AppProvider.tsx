import React, { useReducer } from 'react';

import { AppContext } from '../contexts';
import { ThemeType } from '../theme';
import { User } from '../types';

const AppConsumer = AppContext.Consumer;

interface Action {
  type: 'reset-user' | 'set-user' | 'change-theme-mode';
  payload: any;
}

interface Props {
  navigation?: any;
  children?: any;
}

export interface State {
  user: User;
  theme: ThemeType;
}

export const initialState: State = {
  theme: ThemeType.LIGHT,
  user: {
    displayName: '',
    age: 0,
    job: '',
  },
};

const reducer = (state: State, action: Action) => {
  switch (action.type) {
  case 'change-theme-mode':
    return { ...state, theme: action.payload.theme };
  case 'reset-user':
    return { ...state, user: initialState.user };
  case 'set-user':
    return { ...state, user: action.payload };
  }
};

function AppProvider(props: Props) {
  const [state, dispatch] = useReducer(reducer, initialState);
  const value = { state, dispatch };

  return (
    <AppContext.Provider value={value}>{props.children}</AppContext.Provider>
  );
}

export { AppConsumer, AppProvider, AppContext };
