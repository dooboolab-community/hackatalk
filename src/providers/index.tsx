import * as Device from 'expo-device';

import { ThemeProvider, ThemeType } from '@dooboo-ui/native-theme';
import { dark, light } from '../theme';

import { ApolloProvider } from '@apollo/react-hooks';
import { AuthProvider } from './AuthProvider';
import { DeviceProvider } from './DeviceProvider';
import { FriendProvider } from './FriendProvider';
import { ProfileModalProvider } from './ProfileModalProvider';
import React from 'react';
import { User } from '../types';
import testClient from '../apollo/testClient';

interface Props {
  initialDeviceType?: Device.DeviceType;
  initialThemeType?: ThemeType;
  initialAuthUser?: User;
  children?: React.ReactElement;
}

// hyochan => for testing
export const AllProviders = ({
  initialThemeType,
  initialAuthUser,
  initialDeviceType,
  children,
}: Props): React.ReactElement => {
  return (
    <DeviceProvider
      initialDeviceType={initialDeviceType}
    >
      <ThemeProvider
        initialThemeType={initialThemeType}
        customTheme={{ light, dark }}
      >
        <ApolloProvider client={testClient}>
          <FriendProvider>
            <AuthProvider initialAuthUser={initialAuthUser}>
              <ProfileModalProvider>
                {children}
              </ProfileModalProvider>
            </AuthProvider>
          </FriendProvider>
        </ApolloProvider>
      </ThemeProvider>
    </DeviceProvider>
  );
};
