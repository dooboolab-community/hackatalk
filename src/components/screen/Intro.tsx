import { NavigationScreenProp, NavigationStateRoute } from 'react-navigation';
import React, { useContext, useState } from 'react';

import { AppContext } from '../../providers';
import Button from '../shared/Button';
import { IC_MASK } from '../../utils/Icons';
import { ThemeType } from '../../theme';
import { User } from '../../types';
import { View } from 'react-native';
import { getString } from '../../../STRINGS';
import styled from 'styled-components/native';

const Container = styled.View`
  flex: 1;
  align-self: stretch;
  overflow: scroll;
  background-color: ${({ theme }) => theme.background};

  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  overflow: hidden;
`;

const ContentWrapper = styled.View`
  flex-direction: column;
  height: 100%;
  width: 100%;
  justify-content: flex-start;
  align-items: center;
`;

const ButtonWrapper = styled.View`
  position: absolute;
  flex-direction: column;
  bottom: 40;
  width: 85%;
  align-self: center;
`;

const StyledText = styled.Text`
  font-size: 18;
  line-height: 27;
  color: ${({ theme }) => theme.fontColor};
`;

interface Props {
  store?: any;
  navigation: NavigationScreenProp<NavigationStateRoute<any>>;
}

export default function Intro({ navigation }: Props) {
  let timer: any;
  const { state: { theme, user }, dispatch } = useContext(AppContext);
  const [isLoggingIn, setIsLoggingIn] = useState(false);

  const onLogin = () => {
    dispatch({ type: 'reset-user' });
    setIsLoggingIn(true);
    timer = setTimeout(() => {
      const loggedUser: User = {
        displayName: 'dooboolab',
        age: 30,
        job: 'developer',
      };
      dispatch({ type: 'set-user', payload: loggedUser });
      setIsLoggingIn(false);
      clearTimeout(timer);
    }, 1000);
  };

  const changeTheme = () => {
    let payload: object;
    if (theme === ThemeType.LIGHT) {
      payload = {
        theme: ThemeType.DARK,
      };
    } else {
      payload = {
        theme: ThemeType.LIGHT,
      };
    }
    dispatch({
      type: 'change-theme-mode',
      payload,
    });
  };

  return (
    <Container>
      <ContentWrapper>
        <StyledText
          style={{
            marginTop: 100,
          }}
        >
          {user.displayName}
        </StyledText>
        <StyledText>{user.age ? user.age : ''}</StyledText>
        <StyledText>{user.job}</StyledText>
      </ContentWrapper>
      <ButtonWrapper>
        <Button
          testID='btn1'
          imgLeftSrc={IC_MASK}
          isLoading={isLoggingIn}
          onClick={() => onLogin()}
          text={getString('LOGIN')}
        />
        <View style={{ marginTop: 8 }} />
        <Button
          testID='btn2'
          onClick={() => navigation.navigate('Temp')}
          text={getString('NAVIGATE')}
        />
        <View style={{ marginTop: 8 }} />
        <Button
          testID='btn3'
          onClick={() => changeTheme()}
          text={getString('CHANGE_THEME')}
        />
      </ButtonWrapper>
    </Container>
  );
}
