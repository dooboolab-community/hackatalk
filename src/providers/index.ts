import { AppConsumer, AppContext, AppProvider } from './AppProvider';

export {
  AppProvider,
  AppConsumer,
  AppContext,
};
