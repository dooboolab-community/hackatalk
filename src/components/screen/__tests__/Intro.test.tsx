import 'react-native';

import * as React from 'react';

// Note: test renderer must be required after react-native.
import {
  RenderResult,
  act,
  fireEvent,
  render,
} from '@testing-library/react-native';
import { ThemeType, createTheme } from '../../../theme';

import { AppProvider } from '../../../providers';
import Button from '../../shared/Button';
import Intro from '../Intro';
import { ThemeProvider } from 'styled-components/native';
import renderer from 'react-test-renderer';

const createTestProps = (obj: object) => ({
  navigation: {
    navigate: jest.fn(),
  },
  ...obj,
});

const props: any = createTestProps({});

// test for the container page in dom
describe('[Intro] screen rendering test', () => {
  const component = (
    <AppProvider>
      <ThemeProvider theme={createTheme(ThemeType.LIGHT)}>
        <Intro {...props} />
      </ThemeProvider>
    </AppProvider>
  );
  let json: renderer.ReactTestRendererJSON;

  it('should render outer component and snapshot matches', () => {
    json = renderer.create(component).toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('[Intro] Interaction', () => {
  const component = (
    <AppProvider>
      <ThemeProvider theme={createTheme(ThemeType.LIGHT)}>
        <Intro {...props} />
      </ThemeProvider>
    </AppProvider>
  );

  let rendered: renderer.ReactTestRenderer;
  let root: renderer.ReactTestInstance;
  let testingLib: RenderResult;

  it('should simulate [onLogin] click', () => {
    rendered = renderer.create(component);
    root = rendered.root;
    testingLib = render(component);

    jest.useFakeTimers();
    const buttons = root.findAllByType(Button);
    act(() => {
      fireEvent.press(testingLib.queryByTestId('btn1'));
      expect(setTimeout).toHaveBeenCalledTimes(1);
      // expect(context.dispatch).toHaveBeenCalledWith({ type: 'reset-user' });
      // expect(context.dispatch).toHaveBeenCalledWith({ type: 'set-user' }, expect.any(Object));
      jest.runAllTimers();
    });

    expect(clearTimeout).toHaveBeenCalledTimes(1);
    expect(buttons[0].props.isLoading).toEqual(false); // TODO: test with useState
  });

  it('should simulate [navigate] click', () => {
    rendered = renderer.create(component);
    root = rendered.root;

    // const buttons = root.findAllByType(Button);
    // buttons[1].props.onClick();
    act(() => {
      fireEvent.press(testingLib.getByTestId('btn2'), 'click');
    });
    expect(props.navigation.navigate).toBeCalledWith('Temp');
  });
});
