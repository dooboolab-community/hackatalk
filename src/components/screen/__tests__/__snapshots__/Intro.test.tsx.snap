// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Intro] screen rendering test should render outer component and snapshot matches 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "white",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "flex-start",
        "overflow": "hidden",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "height": "100%",
          "justifyContent": "flex-start",
          "width": "100%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 18,
            "lineHeight": 27,
          },
          Object {
            "marginTop": 100,
          },
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 18,
            "lineHeight": 27,
          },
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 18,
            "lineHeight": 27,
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "bottom": 40,
          "flexDirection": "column",
          "position": "absolute",
          "width": "85%",
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btn1"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#069ccd",
              "borderColor": "#069ccd",
              "borderRadius": 4,
              "borderWidth": 2,
              "height": 52,
              "justifyContent": "center",
              "width": 320,
            },
          ]
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "height": 24,
                "left": 16,
                "position": "absolute",
                "width": 24,
              },
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 14,
              },
            ]
          }
        >
          Login
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": 8,
        }
      }
    />
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btn2"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#069ccd",
              "borderColor": "#069ccd",
              "borderRadius": 4,
              "borderWidth": 2,
              "height": 52,
              "justifyContent": "center",
              "width": 320,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 14,
              },
            ]
          }
        >
          Navigate
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": 8,
        }
      }
    />
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btn3"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#069ccd",
              "borderColor": "#069ccd",
              "borderRadius": 4,
              "borderWidth": 2,
              "height": 52,
              "justifyContent": "center",
              "width": 320,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 14,
              },
            ]
          }
        >
          Change theme
        </Text>
      </View>
    </View>
  </View>
</View>
`;
